tests/mock_model.h

```diff
@@ -102,12 +101,12 @@ class MockModel : public ModelBase<MockModel> {
 };
 
 static inline RegressionDataset<MockFeature>
-mock_training_data(const int n = 10) {
+mock_training_data(const Eigen::Index n = 10) {
   std::vector<MockFeature> features;
   Eigen::VectorXd targets(n);
   for (int i = 0; i < n; i++) {
     features.push_back(MockFeature(i));
-    targets[i] = static_cast<double>(i + n);
+    targets[static_cast<Eigen::Index>(i)] = cast::to_double(i + n);
```

Comment:
`to_index`?

Reply:
Not in this case -- it's not being cast from an STL index (`std::size_t`) but from a mock feature of type `int`.  I can add some more checks for other cases like this, but I tried to focus here mainly on safely handling index conversions.

Reply:
Ah, I see.  Yeah all good, just thought it was an overlooked case.